ztp/gitops-subscriptions/argocd/example/image-based-upgrades/README.md

```diff
@@ -82,12 +83,38 @@ spec:
     foobar-app
 ```
 
-> [!IMPORTANT]
-> Don't forget to create and include the policy [example-oadp-policy.yaml](./example-oadp-policy.yaml) under the `resources` object in the `kustomization.yaml` file.
+Choose either [ibu-upgrade-ranGen.yaml](./ibu-upgrade-ranGen.yaml) example using ZTP `PolicyGenTemplate` or [acm-ibu-upgrade-ranGen.yaml](./acm-ibu-upgrade-ranGen.yaml) example using ACM `PolicyGenerator` to create policies for performing Image Based Upgrades(IBU). Both examples generate the same policies as following:
+* group-ibu-oadp-cm-policy: propogate OADP configmap from hub cluster to target spoke clusters in the `openshift-adp` namespace
+* group-ibu-prep-policy: to transition ibu to Prep stage
+* group-ibu-upgrade-policy: to transition ibu to Upgrade stage
+* group-ibu-finalize-policy: to transition ibu to Idle stage
+* group-ibu-rollback-policy(optional): to transition ibu to Rollback stage
 
+Add the template to [kustomization.yaml](./kustomization.yaml) file in the `generator` object.
+```yaml
+generator:
+# Use policygentemplate to create oadp cm and ibu policies
+- ibu-upgrade-ranGen.yaml
+# Use acmpolicygenerator to create oadp cm and ibu policies
+# - acm-ibu-upgrade-ranGen.yaml
+```
+
+When `ibu-upgrade-ranGen.yaml` is used, override the oadp configmap data field with hub template using the Kustomize patches.
+```yaml
+patches:
+- target:
+    group: policy.open-cluster-management.io
+    version: v1
+    kind: Policy
+    name: group-ibu-oadp-cm-policy
+  patch: |-
+    - op: replace
+      path: /spec/policy-templates/0/objectDefinition/spec/object-templates/0/objectDefinition/data
+      value: '{{hub copyConfigMapData "ztp-group" "oadp-cm" hub}}'
+```
 
-### Enforcing the Policy
+### Enforcing the Policies
 
-To enforce the policy and distribute the ConfigMap to the applicable managed clusters, create a ClusterGroupUpgrade (CGU) CR referencing the aforementioned policy. This should be done prior to setting the IBU stage to `Prep`.
+To enforce the stage policies for performing Image Based Upgrades(IBU), create a ClusterGroupUpgrade (CGU) CR for each stage policy. The `group-ibu-oadp-cm-policy` policy, which distributes the OADP configmap to applicable managed clusters, should be included in the Prep CGU along with `group-ibu-prep-policy`. Since the OADP configmap should be propogated prior to transitioning the IBU stage to `Prep`, it must be the first policy in the Prep CGU.
```

Comment:
```suggestion
To enforce the stage policies for performing Image Based Upgrades (IBU), create a ClusterGroupUpgrade (CGU) CR for each stage policy. The `group-ibu-oadp-cm-policy` policy, which distributes the OADP configmap to applicable managed clusters, should be included in the Prep CGU along with `group-ibu-prep-policy`. Since the OADP configmap should be propagated prior to transitioning the IBU stage to `Prep`, it must be the first policy in the Prep CGU.
```

Reply:
small nit, Image Based Upgrades (IBU) is defined multiple times, just need one time at the beginning of this README, then we could use just IBU, for your consideration

Reply:
@leo8a thanks for your comments! Updated